### PR TITLE
LIMS-4795:Orders: Test plan Approach: In case I select large number of test units in one test plan , they should display successfully in the pop up

### PR DESCRIPTION
**Pull request by omnia353**
```
DevTools listening on ws://127.0.0.1:60491/devtools/browser/f190bad4-b4bb-4dfa-812c-4a0fd162ced5
Orders: Test plan Approach: In case I select large number of test units in one test plan ...
2020-08-27 12:52:17.117 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test069_select_large_number_of_test_units_in_one_testplan
2020-08-27 12:52:21.068 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:568 - wait until page is loaded
2020-08-27 12:52:21.117 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-08-27 12:52:21.762 | INFO     | api_testing.apis.orders_api:set_configuration:412 - set order configuration
2020-08-27 12:52:24.450 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 0
2020-08-27 12:52:24.453 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders
2020-08-27 12:52:25.404 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:25.406 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test069_select_large_number_of_test_units_in_one_testplan:49 - get random order
2020-08-27 12:52:29.081 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:568 - wait until page is loaded
2020-08-27 12:52:30.310 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-08-27 12:52:31.022 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test069_select_large_number_of_test_units_in_one_testplan:52 - create testplan with random data
2020-08-27 12:52:31.024 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/articles
2020-08-27 12:52:31.521 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:31.532 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-08-27 12:52:31.681 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:31.692 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:31.854 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:31.863 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/266
2020-08-27 12:52:32.000 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.002 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:32.154 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.159 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/267
2020-08-27 12:52:32.272 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.275 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:32.449 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.458 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/268
2020-08-27 12:52:32.596 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.605 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:32.818 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.827 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/269
2020-08-27 12:52:32.973 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:32.976 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:33.137 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:33.139 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/270
2020-08-27 12:52:33.243 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:33.253 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:34.040 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:34.043 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/271
2020-08-27 12:52:34.154 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:34.157 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-08-27 12:52:34.294 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:34.297 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/272
2020-08-27 12:52:34.402 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:34.404 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-08-27 12:52:34.595 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-08-27 12:52:34.605 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test069_select_large_number_of_test_units_in_one_testplan:57 - update first suborder
2020-08-27 12:52:35.514 | INFO     | ui_testing.pages.order_page:update_suborder:464 - Set material type : dc0511563f
2020-08-27 12:52:43.031 | INFO     | ui_testing.pages.base_pages:confirm_popup:76 - confirming the popup
2020-08-27 12:52:43.309 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:52:45.176 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:52:56.067 | INFO     | ui_testing.pages.base_pages:confirm_popup:76 - confirming the popup
2020-08-27 12:52:56.087 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:52:57.787 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:52:59.118 | INFO     | ui_testing.pages.order_page:update_suborder:480 -  Set test plan : ['325ee3470a'] for 1 time(s)
2020-08-27 12:53:06.568 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:53:07.926 | INFO     | ui_testing.pages.order_page:update_suborder:489 -  Set test unit : [] for 0 time(s)
2020-08-27 12:53:07.929 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test069_select_large_number_of_test_units_in_one_testplan:61 - get testplan popup
2020-08-27 12:53:08.017 | DEBUG    | ui_testing.pages.base_pages:sleep_small:43 - wait up to 1 sec
2020-08-27 12:53:09.865 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-08-27 12:53:14.224 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 104.905s

OK

```

### CODE REVIEW

File: api_testing/apis/test_plan_api.py [hunk 1]
```diff
@@ -315,16 +315,19 @@ def create_testplan_with_article_not_all(self):
         else:
             raise Exception(f'cant create the test plan with payload {payload}')
 
-    def create_completed_testplan_random_data(self):
+    def create_completed_testplan_random_data(self,no_testplans=1):
```
Comment (ghada-37) [inline api_testing/apis/test_plan_api.py:318]: could you rename `no_testplans` to `no_testunits` to make name clear, you create only one test plan with multiple test units

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -1,43 +1,71 @@
...
+        , they should display successfully in the pop up
+        LIMS-4795
+
+        """
+        orders, payload = self.orders_api.get_all_orders(limit=40)
+        self.info('get random order')
+        random_order = random.choice(orders['orders'])
+        self.orders_page.get_order_edit_page_by_id(random_order['id'])
+        self.info('create testplan with random data')
+        payload = TestPlanAPI().create_completed_testplan_random_data(no_testplans=7)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:53]: you need to create test plan before open edit page of order, to let test plan creation reflect on order page

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py:53]: Also, rename it to test_plan = .. as payload variable you used above .

Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

File: api_testing/apis/test_plan_api.py (hunk 1 above)
Comment (omnia353) [reply api_testing/apis/test_plan_api.py]: done

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 3]
```diff
@@ -1,43 +1,71 @@
...
+        self.orders_api.set_configuration()
+
+    def test069_select_large_number_of_test_units_in_one_testplan(self):
+        """
+        Orders: Test plan Approach: In case I select large number of test units in one test plan
+        , they should display successfully in the pop up
+        LIMS-4795
+
+        """
+        orders, payload = self.orders_api.get_all_orders(limit=40)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:48]: replace this with :
`order = self.orders_api.get_order_with_multiple_sub_orders()`
as in description of test case it say: try this from 3rd suborder

**Review by ghada-37 — CHANGES_REQUESTED** [review]

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 3 above)
Comment (omnia353) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

**Review by ghada-37 — APPROVED** [review]

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 4]
```diff
@@ -2871,3 +2871,29 @@ def test084_filter_by_changed_by(self):
...
+        self.orders_page.get_order_edit_page_by_id(order['id'])
+        testunit_names = []
+        for testunit in testPlan[1]['testUnits']:
+            testunit_names.append(testunit['name'])
+        self.info('update third suborder')
+        self.order_page.update_suborder(sub_order_index=2, material_type=testPlan[1]['materialType'][0]['text'],
+                                        articles=[testPlan[1]['selectedArticles'][0]['text']],
+                                        test_plans=[testPlan[1]['testPlan']['text']])
+        self.info('get testplan popup')
+        results = self.order_page.get_testplan_pop_up()
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2895]: @omnia353 I found you make         results = self.order_page.get_testplan_pop_up()
but you didn't selected which test plan pop-up you need to check, 
So, it always checks first test plan popup

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 5]
```diff
@@ -2871,3 +2871,29 @@ def test084_filter_by_changed_by(self):
...
+        for result in results:
+            if result['test_plan'] == testPlan[1]['testPlan']['text']:
+                for testunit in testunit_names:
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2898]: 2) you need to get test units in pop up in list and make sure len = 7, and they are the same in test plan

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 6]
```diff
@@ -2871,3 +2871,29 @@ def test084_filter_by_changed_by(self):
...
+        order = self.orders_api.get_order_with_multiple_sub_orders(no_suborders=2)
+        self.info('create testplan with random data')
+        testPlan = TestPlanAPI().create_completed_testplan_random_data(no_testunits=7)
+        self.info('get random order')
+        self.orders_page.get_order_edit_page_by_id(order['id'])
+        testunit_names = []
+        for testunit in testPlan[1]['testUnits']:
+            testunit_names.append(testunit['name'])
+        self.info('update third suborder')
+        self.order_page.update_suborder(sub_order_index=2, material_type=testPlan[1]['materialType'][0]['text'],
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:2891]: use create new suborder not update, to make sure that you have many suborders

**Review by ghada-37 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**